core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java

```diff
@@ -572,6 +577,28 @@ private void catchUp(JournalStateMachine stateMachine, LocalFirstRaftClient clie
     // Wait for any outstanding snapshot to complete.
     CommonUtils.waitFor("snapshotting to finish", () -> !stateMachine.isSnapshotting(),
         WaitForOptions.defaults().setTimeoutMs(10 * Constants.MINUTE_MS));
+    OptionalLong endCommitIndex = OptionalLong.empty();
+    try {
+      // raft peer IDs are unique, so there should really only ever be one result.
+      // If for some reason there is more than one..it should be fine as it only
+      // affects the completion time estimate in the logs.
+      Optional<RaftProtos.CommitInfoProto> commitInfo = getGroupInfo().getCommitInfos().stream()
+              .filter(commit ->
+                  mServer.getId().equals(RaftPeerId.valueOf(commit.getServer().getId())))
+              .findFirst();
+      if (commitInfo.isPresent()) {
+        endCommitIndex = OptionalLong.of(commitInfo.get().getCommitIndex());
+      }
```

Comment:
(nit) shall we log a warning if `commitInfo` is not present?

Reply:
updated